GunShipAlpha/src/Enemy.java

```diff
@@ -38,4 +78,17 @@ public boolean isIntersectingLine(LineLogic line) {
     private Rectangle createBounds() {
         return new Rectangle(x, y, EnemySprite.width(), EnemySprite.height());
     }
+    public Projectile fireAtPlayer() {
+        double targetX = 400;
+        double targetY = 400;
+        double directionX = targetX - this.x;
+        double directionY = targetY - this.y;
+        double magnitude = Math.sqrt(directionX * directionX + directionY * directionY);
+        directionX /= magnitude;
+        directionY /= magnitude;
+        double offsetX = directionX * EnemySprite.width() / 2;
+        double offsetY = directionY * EnemySprite.height() / 2;
+        return new Projectile((int) (this.x + offsetX), (int) (this.y + offsetY), new Point2D.Double(directionX, directionY));
```

Comment:
it feels like this whole function could be expressed more clearly as a few pieces. there's a function that wants to be here like "vectorBetween(enemy, player)" and then the projectile is "new Projectile(location, vector)"

Reply:
fixed